Connect User History to DB

ChenJesse opened the pull request:
Fixes the API for fetching post history, makes calendar date (on ProfileViewController) red if user posted on that day

Maybe to be implemented later, shade of red varies with how many likes the post received?

https://github.com/cuappdev/icefishing/issues/38

lucasderraugh commented:
The API change is fine (thanks for fixing it). The history comparison stuff though should not be compared via Strings. @acheng96 took the liberty to make an NSDate extension that returns the current day given a date. See [NSDateExtension.swift](https://github.com/cuappdev/icefishing/blob/master/IceFishing/Profile/NSDateExtension.swift)

ChenJesse commented:
OK, I fixed some more bugs and added some implementation of the changing color by changing transparency.

Also, I didn't realize that after you push to remote, squashing commits on local won't change the commit history on the remote; is there any way to deal with that?

lucasderraugh commented:
It should match whatever you have in `git log`. However, don't use squash or fixup until you push to master, that's where I would like it to look simplified, doesn't matter on this branch.

lucasderraugh commented:
When selecting one of the history cells, it should transition to a view that has the post for that day.

lucasderraugh commented on IceFishing/Controllers/PostHistoryTableViewController.swift:
```diff
@@ -7,22 +7,41 @@
+	
+	var posts: [Post] = []
+	var songLikes: [Int] = []
     var postedDates: [NSDate]! = []
```
Implicitly unwrapped `postedDates` unnecessary.

lucasderraugh commented on IceFishing/Controllers/PostHistoryTableViewController.swift:
```diff
@@ -48,29 +72,22 @@ class PostHistoryTableViewController: UITableViewController {
     
     override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return self.postedDates.count
+        return self.posts.count
```
No self

lucasderraugh commented on IceFishing/Controllers/PostHistoryTableViewController.swift:
```diff
@@ -48,29 +72,22 @@ class PostHistoryTableViewController: UITableViewController {
+		cell.postView.post = posts[indexPath.row]
+		cell.postView.post?.player.prepareToPlay()
+		
         let formatter = NSDateFormatter()
```
Formatters are expensive! Reuse by adding one by adding it to `NSDateFormatter+Shared.swift` if needed.

lucasderraugh commented on IceFishing/Controllers/PostHistoryTableViewController.swift:
```diff
@@ -78,8 +95,150 @@ class PostHistoryTableViewController: UITableViewController {
     override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
-        let selectedCell = tableView.cellForRowAtIndexPath(indexPath)!
-        selectedCell.contentView.backgroundColor = UIColor.iceLightGray
+        let selectedCell = tableView.cellForRowAtIndexPath(indexPath)! as! FeedTableViewCell
```
Redundant ! after `cellForRow` call

lucasderraugh commented on IceFishing/Profile/ProfileViewController.swift:
```diff
@@ -34,9 +38,16 @@ class ProfileViewController: UIViewController, UICollectionViewDelegate, UIColle
         super.viewDidLoad()
         
         // TODO: Backend route is currently wrong
```
Remove this `TODO`, since it is now fixed.

lucasderraugh commented on IceFishing/Profile/ProfileViewController.swift:
```diff
@@ -143,19 +154,43 @@ class ProfileViewController: UIViewController, UICollectionViewDelegate, UIColle
+	private func determineAlpha(likes: Int) -> CGFloat {
+		let ratio = Float(likes) / avgLikes
+		
+		if (ratio <= 0.2) {
```
Would look nicer if it was more of a functional mapping to the value, perhaps with some caps, instead of lots of `else if` statements.

lucasderraugh commented on IceFishing/Profile/ProfileViewController.swift:
```diff
@@ -143,19 +154,43 @@ class ProfileViewController: UIViewController, UICollectionViewDelegate, UIColle
-        if postedDates.contains(cell.date) {
-            cell.dayInnerCircleView.backgroundColor = UIColor.iceDarkRed
+		
+		if let index = postedDays.indexOf(cell.date.day()) where (cell.date.month() == postedDates[index].month() && cell.date.year() == postedDates[index].year()) {
```
Remove unnecessary parenthesis after `where`.

lucasderraugh commented on IceFishing/Views/FeedTableViewCell.swift:
```diff
@@ -12,9 +12,15 @@ class FeedTableViewCell: UITableViewCell {
 	
     @IBAction func saveButtonClicked(sender: UIButton) {
-        referenceFeedViewController.saveButtonClicked()
+		if let viewController = referenceFeedViewController {
```
If you could take a bit of time to chat with me on the changes here that would be great!

lucasderraugh commented:
Aside from the comments provided, it looks pretty good.

lucasderraugh commented:
Checked over in Wednesday session. Shipping new code.
